compiler/rustc_ast/src/visitors.rs

```diff
@@ -481,7 +496,32 @@ macro_rules! make_ast_visitor {
             fn visit_fn_decl(&mut self, d: ref_t!(P!(FnDecl))) -> result!() {
                 walk_fn_decl(self, d)
             }
+        }
+
+        macro_rules! visit_span {
+            ($vis: ident, $span: ident) => {
+                if_mut_stmt!(
+                    $vis.visit_span($span)
+                ,
+                    // assign to _ to prevent unused_variable warnings
+                    let _ = (&$vis, &$span)
+                )
```

Comment:
Should we just have `visit_span` in both visitors?

Reply:
In the original issue, @oli-obk mentioned that visiting ids and spans doesn't really make sense for Visitors, then I didn't add `visit_span` and `visit_id` for Visitors and I came up with this.

Reply:
We can try it out in a follow-up if there are any use cases for it.